### ISSUE TITLE
[P1] server 重啟會重用 param_version，PPO 把舊牌譜配到新權重而 ratio 檢查仍通過

### ISSUE THREAD

**Issue by HHim8826** [HHim8826/Mortal#8]
審查分支：`train-parquet`，版本 `08543729000846d56c2602251245596c27f4b182`。這是 `c623f37` 的版本化 replay、`12af353` 的歷史快照恢復及 `75bdb91` 的 identity check 之間的生命週期缺口。

[server 重新初始化版本：server.py:183–188](https://github.com/HHim8826/Mortal/blob/08543729000846d56c2602251245596c27f4b182/mortal/server.py#L183-L188)；[發布時僅遞增記憶體計數器：123–134](https://github.com/HHim8826/Mortal/blob/08543729000846d56c2602251245596c27f4b182/mortal/server.py#L123-L134)；[舊 worker 提交版本直接採用：108–117](https://github.com/HHim8826/Mortal/blob/08543729000846d56c2602251245596c27f4b182/mortal/server.py#L108-L117)；[Behaviour 覆寫同號快照：train_ppo.py:123–132](https://github.com/HHim8826/Mortal/blob/08543729000846d56c2602251245596c27f4b182/mortal/train_ppo.py#L123-L132)。

param_version 每次 server 啟動都從 0 開始，但 worker 的在途遊戲和 trainer 的 params/vN.pth 可以跨越 server 的生命週期。新 server 再發布 v1、v2… 時，Behaviour.add() 會直接把同號的舊 head／檔案換成新權重。舊 worker 隨後提交的 vN 仍被當成「已知版本」，從新 head 重算 denominator；没有 session ID、內容 hash 或碰撞檢查可阻止這件事。僅清空 server 的 buffer/drain 不會取消 worker 已在進行的遊戲。

已離線呼叫實際 State、Handler.handle_submit_param/handle_submit_replay 和 Behaviour，以重建 State 模擬 server restart：
```text
舊 server 發布：v1，P_old(action=0)=0.9
server restart
新 server 發布：v1，P_new(action=0)=0.1
舊 worker 提交：0_v1_1_2827_0.json.gz

實際 behaviour probability：   0.899999976
歷史查詢得到的 probability：  0.099999994
程式使用的 ratio：             1.0
正確 ratio：                   0.111111107
Behaviour.matches(...)：       True
```

ratio=1 的檢查比較的是「現在 head」和「同號但已被覆寫的 head」，兩邊一起錯配時必然通過，不能證明這些權重真的產生過該 replay。這會直接污染 PPO importance weighting，不只是丟掉幾筆舊資料。

建議：讓參數身分跨 server restart 仍唯一，例如持久化且單調遞增的版本，或 session UUID + counter / checkpoint hash，並在 worker、檔名與快照中完整傳遞。版本碰撞應拒絕或明確隔離；必要時以 replay 記錄的 logits 做獨立一致性驗證。
